decimal.go

```diff
@@ -967,10 +967,7 @@ func (d Decimal) IntPart() int64 {
 
 // BigInt returns integer component of the decimal as a BigInt.
 func (d Decimal) BigInt() *big.Int {
-	scaledD := d.rescale(0)
-	i := &big.Int{}
-	i.SetString(scaledD.String(), 10)
-	return i
+	return new(big.Int).Set(d.value)
```

Comment:
```suggestion
	scaledD := d.rescale(0)
	return new(big.Int).Set(scaledD.value)
```
@mwoss this corrects your feedback. btw, why does rescale need to return a new Decimal when `exp == d.exp`? would expect with non-mutable API it'd be fine to `return d`

Reply:
Created #359

Reply:
Thanks for working on that :D 

---

> btw, why does rescale need to return a new Decimal when exp == d.exp? would expect with non-mutable API it'd be fine to return d

If I remember correctly we have done it because, in a few places in the codebase, we expect that `rescale` always returns a new decimal, so we can safely perform operations on `d.value`. It could be improved for sure, but we would need to track down all those places and safely update the code and docs.